crates/net/network/src/peers/manager.rs

```diff
@@ -298,8 +298,18 @@ impl PeersManager {
 
     /// Bans the peer temporarily with the configured ban timeout
     fn ban_peer(&mut self, peer_id: PeerId) {
-        self.ban_list.ban_peer_until(peer_id, std::time::Instant::now() + self.ban_duration);
-        self.queued_actions.push_back(PeerAction::BanPeer { peer_id });
+        if let Some(peer) = self.peers.get(&peer_id) {
+            if peer.is_trusted() {
+                self.ban_list.ban_peer_until(
+                    peer_id,
+                    std::time::Instant::now() + self.backoff_durations.medium,
```

Comment:
where did `self.backoff_durations.medium` come from? isn't it supposed to use a fraction of `self.ban_duration`?

Reply:
As discussed with @mattsse

Reply:
Then will let @mattsse continue review :)